engine/src/test/java/org/pentaho/di/job/JobMetaTest.java

```diff
@@ -500,4 +500,13 @@ public void testUpdateCurrentDirWithoutFilenameOrRepository( ) {
     assertEquals( "Original value defined at run execution", jobMetaTest.getVariable( Const.INTERNAL_VARIABLE_ENTRY_CURRENT_DIRECTORY )  );
   }
 
+  @Test
+  public void isGatheringMetricsTest() {

```

Comment:
In my opinion, this kind of test should be use reflection to set/get the value.

Example:

getA() { return b; }
setA( a ) { b = a; }

The previous example shows how we can be fooled by own implementation and the unit test doesn't catch it.